### DISCUSSION TITLE
get 접근자를 쓰는 것과 쓰지 않는 것

### DISCUSSION THREAD

**Discussion by juunzzi** [woowacourse-study/2022-book-muscle#39] [동키콩]
https://github.com/woowacourse-study/2022-book-muscle/pull/33#discussion_r819391864

**Comment by JUDONGHYEOK** [comment]
https://stackoverflow.com/questions/53248298/difference-between-using-setters-or-methods-in-javascript
자스민과 페어를 하다가 비슷한 주제가 나와서 찾은 글 공유할게요!! 이 글에 의하면 setter는 기술적으로 큰 차이는 없지만 접근방법에 따른 의미상의 차이라고 보는듯 합니다!

**Comment by JUDONGHYEOK** [comment]
https://martinfowler.com/bliki/UniformAccessPrinciple.html

리팩터링 책을 읽다 또 발견 한 건데요. 이 책의 저자인 마틴 파울러가 쓴 단일접근 원칙에 관한 글입니다. 
> 이 원칙의 핵심은 person 객체가 있고, 그 객체의 나이값을 알고 싶을 때 나이가 객체 내의 필드에 저장되어 있든 계산된 값이든 상관없이 같은 방식으로 접근해야 한다는 것이다.

즉 예를 들어 person 클래스에 나이에 대한 필드가 있다고 가정하고 나이에 대한 접근자와 이 나이를 통해 계산한 값을 리턴한 값을 동일하게 접근해야한다는 주장입니다.

http://egloos.zum.com/aeternum/v/3028192

위 글을 인용한 인스턴트 푸드같은 블로그 글인데요. 위 글에서도 언어마다 특징은 다르겠지만 다음과 같이 추천을 하고 있네요

> 속성 대신 명령/질의(COMMAND/QUERY)을 사용할 경우 코드의 가독성이 저하되고 복잡도가 상승하게 된다. customer.mileage += 10에 비해 customer.setMileage(customer.getMileage() + 10)을 이해하는데 필요한 인지 과부하가 더 크다

method를 통한 접근 보다 getter setter를 사용하는 이유는 아무래도 의미상의 이유가 가장 큰 것 같아요. 마지막으로 공식 문서는 아니지만 w3cschool에서 말하는 getter setter를 사용하는 이유에 관한 글을 첨부하겠습니다!!

https://www.w3schools.com/js/js_object_accessors.asp

> Why Using Getters and Setters?
It gives simpler syntax
It allows equal syntax for properties and methods
It can secure better data quality
It is useful for doing things behind-the-scenes
